### ISSUE TITLE
[Bug] cross-compile tcpreplay for armv5 freescale

### ISSUE THREAD

**Issue by dummys** [appneta/tcpreplay#623]
Hi,

Im trying to cross compile tcpreplay for armv5 but I get the following eror when doing it:
PS: libpcap is compiled successfully. I'm using two different toolchain, one from crosstool-ng `rm-unknown-linux-gnueabi` and `arm-unknown-linux-gnueabihf` but both fail with the same error: 

```
In file included from ../../src/common/timer.h:31,
                 from ../../src/common.h:45,
                 from cidr.c:23:
/usr/include/math.h:462:21: error: '_Float128' is not supported on this target
 # define _Mdouble_  _Float128
```

My command is: 

```
CC=arm-unknown-linux-gnueabi-gcc ./configure --host=arm-unknown-linux-gnueabi --with-libpcap=/home/build/cross-compiling/arm-32/libpcap-1.9.1_comp/compiled --prefix=/home/build/cross-compiling/arm-32/tcpreplay-4.3.3/compiled
```

**Comment by fklassen** [comment]
Not sure why it is picking up the local math.h rather than the cross compiler's. Maybe get rid of `--host=`

Can you compile 64-bit?

**Comment by dummys** [comment]
no I can't because the target is x32

**Comment by dummys** [comment]
@fklassen do i need to cross compile tcpdump and libdnet too ?

**Comment by dummys** [comment]
i cross compiled libdnet, tcpdump also and use this now:

```
set -x PATH "/home/build/x-tools/arm-unknown-linux-gnueabi/bin/:/home/mofo/x-tools/arm-unknown-linux-gnueabi/arm-unknown-linux-gnueabi/sysroot/usr/include:$PATH"


set -x CPPFLAGS "/home/build/cross-compiling/arm-32/libpcap-1.9.1/compiled/lib/libpcap.so"


set -x LDFLAGS "/home/build/cross-compiling/arm-32/libpcap-1.9.1/compiled/lib/libpcap.so"

./configure --host=arm-unknown-linux-gnueabi --with-libpcap=/home/build/cross-compiling/arm-32/libpcap-1.9.1 --prefix=/home/build/cross-compiling/arm-32/tcpreplay-4.3.3/compiled --enable-static-link --with-tcpdump=/home/build/cross-compiling/arm-32/tcpdump-4.9.3/tcpdump --with-libdnet=/home/build/cross-compiling/arm-32/libdnet-1.11/compiled
```

And i get further the error of fpu error, but now I have another one popping:

```
/home/build/x-tools/arm-unknown-linux-gnueabi/lib/gcc/arm-unknown-linux-gnueabi/8.3.0/../../../../arm-unknown-linux-gnueabi/bin/ld.bfd: ./common/libcommon.a(utils.o): in function `_our_safe_realloc':
utils.c:(.text+0xb4): undefined reference to `rpl_realloc'
/home/build/x-tools/arm-unknown-linux-gnueabi/lib/gcc/arm-unknown-linux-gnueabi/8.3.0/../../../../arm-unknown-linux-gnueabi/bin/ld.bfd: ../libopts/.libs/libopts.a(libopts_la-libopts.o): in function `ao_realloc':
libopts.c:(.text+0x2088): undefined reference to `rpl_realloc'
collect2: error: ld returned 1 exit status
make[3]: *** [Makefile:717: tcpreplay] Error 1
```

**Comment by dummys** [comment]
now I struggle with building a static tcpreplay version, its not possible despite I used "-static" everywhere I can, like in lddflags, cflags etc...

**Comment by fklassen** [comment]
It sounds like you found a fix similar to [this](https://github.com/rgerhards/libmaxminddb/commit/d0cd6bec3acce2b8a1af80645410e47bb9ae8145).

You seem to have issues with your compiler. Unfortunately I would say you are on your won to figure out how to get static to work. You probably have issues with your libraries not containing static libraries.

**Comment by dummys** [comment]
yes but the problem of `AC_FUNC_REALLOC` is in your configure.ac. Not in my toolchain. It seems to be a error that appear mainly on cross compile for arm.

**Comment by fklassen** [comment]
fixed in PR #641
